### PR TITLE
Your commit message here

### PR DESCRIPTION
**Pull request by Sou0602**

### CODE REVIEW

File: eval_commavq_ort.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
+import logging
```
Comment (sashank-tirumala) [inline eval_commavq_ort.py:1]: Add a docstring for the file explaining what this file is supposed to do

File: eval_commavq_ort.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+import torch
+import torch.nn.functional as F
+from datasets import load_dataset
+import numpy as np
+from tqdm import tqdm as tqdm
+from model import GPT, GPTConfig
+import pdb
+import os
+import time
```
Comment (sashank-tirumala) [inline eval_commavq_ort.py:18]: Sort your imports using isort

File: eval_commavq_ort.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+# system
+device = 'cuda' # examples: 'cpu', 'cuda', 'cuda:0', 'cuda:1' etc., or try 'mps' on macbooks
+compile = True 
+seed_offset = 0
+torch.manual_seed(1337 + seed_offset)
+torch.backends.cuda.matmul.allow_tf32 = True # allow tf32 on matmul
+torch.backends.cudnn.allow_tf32 = True # allow tf32 on cudnn
+
```
Comment (sashank-tirumala) [inline eval_commavq_ort.py:26]: All the above lines should go inside main

File: eval_commavq_ort.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+import torch.nn.functional as F
+from datasets import load_dataset
+import numpy as np
+from tqdm import tqdm as tqdm
+from model import GPT, GPTConfig
+import pdb
+import os
+import time
+# system
+device = 'cuda' # examples: 'cpu', 'cuda', 'cuda:0', 'cuda:1' etc., or try 'mps' on macbooks
```
Comment (sashank-tirumala) [inline eval_commavq_ort.py:20]: This should be in config or have logic like if cuda exists use cuda else use cpu

File: eval_commavq_ort.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+compile = True
```
Comment (sashank-tirumala) [inline eval_commavq_ort.py:21]: put in config

File: eval_commavq_ort.py [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+from tqdm import tqdm as tqdm
+from model import GPT, GPTConfig
+import pdb
+import os
+import time
+# system
+device = 'cuda' # examples: 'cpu', 'cuda', 'cuda:0', 'cuda:1' etc., or try 'mps' on macbooks
+compile = True 
+seed_offset = 0
+torch.manual_seed(1337 + seed_offset)
```
Comment (sashank-tirumala) [inline eval_commavq_ort.py:23]: Is this needed, just choose some random seed?

File: eval_commavq_ort.py [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+torch.backends.cuda.matmul.allow_tf32 = True # allow tf32 on matmul
+torch.backends.cudnn.allow_tf32 = True # allow tf32 on cudnn
+
+#----------------------------------------------------------------------------------------------------------
+# Getting eval_indices similar to the eval.ipynb file.
+
+## constants 
+BOS_TOKEN = 1024
```
Comment (sashank-tirumala) [inline eval_commavq_ort.py:31]: should go in config or a seperate file called constants, dont make it a global variable

File: eval_commavq_ort.py [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+TOKENS_PER_FRAME = 129
+BS = 10
+CONTEXT_SIZE_FRAMES = 20
+N_FRAMES = 1200
+N = N_FRAMES - 20
```
Comment (sashank-tirumala) [inline eval_commavq_ort.py:36]: Same as above

File: eval_commavq_ort.py [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+
+def main():
+    
+    files,indices = get_eval_files_indices()
+    ort_path = '/home/sudatha/rob/nanoGPT/out-commavq_2/out-commavq_2.onnx'
```
Comment (sashank-tirumala) [inline eval_commavq_ort.py:41]: should go to config

File: eval_commavq_ort.py [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+    sess_options = onnxruntime.SessionOptions()
+    sess_options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_ENABLE_EXTENDED
+    sess_options.add_session_config_entry('session.dynamic_block_base', '4')
```
Comment (sashank-tirumala) [inline eval_commavq_ort.py:44]: maybe add a comment what this is doing?

File: eval_commavq_ort.py [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+
+    provider = 'CUDAExecutionProvider'
+    ort_session = onnxruntime.InferenceSession(Path(ort_path).with_suffix('.onnx'),sess_options,providers=[provider])
+    binding = ort_session.io_binding()
+
+    total_losses = []
+    losses, sizes = [], []
+    pbar = tqdm(files)
+    for f in pbar:
```
Comment (sashank-tirumala) [inline eval_commavq_ort.py:53]: nit: I would prefer `for f in tqdm(files)` over creating a seperate vaiable

File: eval_commavq_ort.py [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+
+    Returns:
+        list: List of file paths.
+        list: List of data indices for slicing.
+    """
+    ds = load_dataset(data_path, num_proc=40)
+    files = ds["40"]["path"]  # 40 is the validation set
+
+    # Create data slicing
+    indices = np.arange(0, N * TOKENS_PER_FRAME)
```
Comment (sashank-tirumala) [inline eval_commavq_ort.py:82]: If those constants are being used this way, then it might make more sense to pass it as an argument to this function. Do we need them as global constants?

File: eval_commavq_ort.py [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+                loss, size = compute_loss(ort_session,binding, tokens, ii)  
+                losses.append(loss)
+                sizes.append(size)
+
+    total_loss = np.sum(losses)/np.sum(sizes)
+    total_losses.append(total_loss)
+    print(f"total loss {np.mean(total_losses)}")
+
+
+def get_eval_files_indices(data_path="commaai/commavq"):
```
Comment (sashank-tirumala) [inline eval_commavq_ort.py:67]: Type Hints would be helpful for example `def get_eval_files_indices(data_path : Path ="commaai/commavq") -> List`

File: export_to_onnx.py [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+from model import GPT,GPTConfig
+
+import torch
+import torch.nn.functional as F
+from datasets import load_dataset
+import numpy as np
+from tqdm import tqdm as tqdm
+from model import GPT, GPTConfig
+import pdb
+import os
```
Comment (sashank-tirumala) [inline export_to_onnx.py:20]: you might want to sort imports with isort

File: export_to_onnx.py [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+    torch.backends.cuda.matmul.allow_tf32 = True # allow tf32 on matmul
+    torch.backends.cudnn.allow_tf32 = True # allow tf32 on cudnn
+
+    ## Model Initialization parameters
+    block_size = 2580
+    n_layer = 8
+    n_head = 8
+    n_embd = 512
+    dropout = 0.2 
+    bias = False
```
Comment (sashank-tirumala) [inline export_to_onnx.py:38]: all this should be in a config I think or command line arguments

File: export_to_onnx.py [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+
+
+if __name__ == "__main__":
+    # system
+    device = 'cuda' # examples: 'cpu', 'cuda', 'cuda:0', 'cuda:1' etc., or try 'mps' on macbooks
+    compile = True 
+    seed_offset = 0
+    torch.manual_seed(1337 + seed_offset)
+    torch.backends.cuda.matmul.allow_tf32 = True # allow tf32 on matmul
+    torch.backends.cudnn.allow_tf32 = True # allow tf32 on cudnn
```
Comment (sashank-tirumala) [inline export_to_onnx.py:30]: same comments as previous file

File: export_to_onnx.py [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+
+    ## Model Initialization parameters
+    block_size = 2580
+    n_layer = 8
+    n_head = 8
+    n_embd = 512
+    dropout = 0.2 
+    bias = False 
+
+    model_args = dict(n_layer=n_layer, n_head=n_head, n_embd=n_embd, block_size=block_size,
```
Comment (sashank-tirumala) [inline export_to_onnx.py:40]: line seems too long, did you use black to format it?

File: export_to_onnx.py [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+                    bias=bias, vocab_size=None, dropout=dropout) 
+
+    ckpt_path =  '/home/sudatha/rob/nanoGPT/out-commavq_2/ckpt.pt'
```
Comment (sashank-tirumala) [inline export_to_onnx.py:43]: goes to config don't put any absolute paths in your code

File: export_to_onnx.py [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+
+
+    checkpoint = torch.load(ckpt_path, map_location=device)
+    checkpoint_model_args = checkpoint['model_args']
+    # force these config attributes to be equal otherwise we can't even resume training
+    # the rest of the attributes (e.g. dropout) can stay as desired from command line
+    for k in ['n_layer', 'n_head', 'n_embd', 'block_size', 'bias', 'vocab_size']:
```
Comment (sashank-tirumala) [inline export_to_onnx.py:50]: can this list be defined in the model, or ina config etc

**Review by sashank-tirumala — COMMENTED** [review]
Initial review, just asking for formatting changes and basic stuff like removing constants/ global variables/ absolute paths
